### DISCUSSION TITLE
`git log master..origin/master -- path/to/A path/to/B`

### DISCUSSION THREAD

**Discussion by zimoun** [magit/magit#4661] [Support] [answered]
Hi,

Usually, I do not pull but instead I fetch and check the new commits touching the part I am interested in, then I rebase.  Therefore, I am using something which looks like:
```
git log master..origin/master -- path/to/A path/to/B
```
And I am not able to have something similar with Magit.  Because I can set only one path where I have many more.  Do I miss something?

Then, I would like to have an Emacs shortcut for running something as above and open a `magit-log` buffer.  How could I do that?

Thanks in adavance.

Cheers,
simon

**Comment by tarsius** [comment, marked as answer]
> Usually, I do not pull but instead I fetch and check the new commits touching the part I am interested in, then I rebase. 

That's what I do, though I don't bother with "touching the part I am interested in".

> And I am not able to have something similar with Magit. Because I can set only one path where I have many more. Do I miss something?

At the `Limit to file(s): ` prompt you can provide multiple files, separated by commas.

> Then, I would like to have an Emacs shortcut for running something as above and open a magit-log buffer. How could I do that?

You could use `magit-post-refresh-hook`, adding a function that checks the value of `this-command`, which then runs `magit-log-setup-buffer` with the appropriate arguments.

You can also set the file filter in the status buffer by pressing `L - -`.

**Reply by zimoun** [reply]
Hi,

Thanks for the reply.

> That's what I do, though I don't bother with "touching the part I am interested in".

Well, considering a repo with many many changes per day, then the log is becoming too large to find my bits between each of my fetches. :-)

> At the `Limit to file(s):` prompt you can provide multiple files, separated by commas.

Cool!  I have missed the comma separator instead of usual space separator.  Well, maybe I missed it, but is this comma separator documented?  Maybe ` Limit to files, comma separated (--) ` would help.  Because it is counter intuitive with the regular Git CLI.  I mean, since the Git CLI is between parenthesis, I was expecting the same separator.

> You can also set the file filter in the status buffer by pressing `L - -`.

I am missing this one.  I am using Magit 3.3.0 with Git 2.34.0 on Emacs 27.2.

Cheers,
simon

**Reply by tarsius** [reply]
> > At the `Limit to file(s):` prompt you can provide multiple files, separated by commas.
> 
> Cool! I have missed the comma separator instead of usual space separator. Well, maybe I missed it, but is this comma separator documented? Maybe `Limit to files, comma separated (--)` would help. Because it is counter intuitive with the regular Git CLI. I mean, since the Git CLI is between parenthesis, I was expecting the same separator.

I went with `Limit to file,s:` like we already do with other multi-value prompts. I choose `,` because that appears less often in file-names (and other completion candidates) than a space.

> > You can also set the file filter in the status buffer by pressing `L - -`.
> 
> I am missing this one.

Yes, turns out that is only supported for logs but not diffs. I intend to keep it that way.
